.github/workflows/cli_commands.yml

```diff
@@ -15,236 +15,344 @@ jobs:
+                    echo "❌ Experimental command verification failed: stable.wasm should not have been created"
+                    exit 1
+                  else
+                    echo "✅ Experimental command verification successful: stable.wasm was correctly not created"
+                  fi
+
+                  # Clean templates for final test
+                  clean_templates
 
                   # Test azle dev template --all
-                  echo "Testing 'azle dev template --all' command"
+                  echo "Testing 'azle dev template --all' command (both templates)"
                   npx azle dev template --all
 
-                  # Verify both templates were created
-                  if [[ -f "dist/canister_templates/stable.wasm" && -f "dist/canister_templates/experimental.wasm" ]]; then
-                    echo "✅ All templates verification successful: Both stable.wasm and experimental.wasm were created"
+                  # Verify both templates were created and have valid sizes
+                  STABLE_EXISTS=false
+                  EXPERIMENTAL_EXISTS=false
+
+                  if [[ -f "dist/canister_templates/stable.wasm" ]]; then
+                    echo "✅ Stable template file created successfully"
+                    if check_file_size "dist/canister_templates/stable.wasm"; then
+                      STABLE_EXISTS=true
+                    else
+                      echo "❌ Stable template size check failed during --all test"
+                      exit 1
+                    fi
                   else
-                    echo "❌ All templates verification failed: Not all template files were created"
-                    echo "stable.wasm exists: $(test -f dist/canister_templates/stable.wasm && echo 'yes' || echo 'no')"
-                    echo "experimental.wasm exists: $(test -f dist/canister_templates/experimental.wasm && echo 'yes' || echo 'no')"
+                    echo "❌ All templates verification failed: stable.wasm was not created"
+                    exit 1
+                  fi
+
+                  if [[ -f "dist/canister_templates/experimental.wasm" ]]; then
+                    echo "✅ Experimental template file created successfully"
+                    if check_file_size "dist/canister_templates/experimental.wasm"; then
+                      EXPERIMENTAL_EXISTS=true
+                    else
+                      echo "❌ Experimental template size check failed during --all test"
+                      exit 1
+                    fi
+                  else
+                    echo "❌ All templates verification failed: experimental.wasm was not created"
+                    exit 1
+                  fi
+
+                  if [[ "$STABLE_EXISTS" == true && "$EXPERIMENTAL_EXISTS" == true ]]; then
+                    echo "✅ All templates verification successful: Both stable.wasm and experimental.wasm were created with valid sizes"
+                  else
```

Comment:
Not sure this last case could ever be hit, thus not sure the `STABLE_EXISTS` and other variable are needed